src/main/ruby/core/posix.rb

```diff
@@ -250,6 +250,7 @@ def self.read_string(fd, length)
         break
       end
       bytes_read += ret
+      break if nonblock
```

Comment:
Working my way through more of the review, I think my concern was addressed in 97ba6de144f0f8c851c7bc26679a37f0cbf9963d.

Reply:
> My understanding of non-blocking `read` could be wrong, but shouldn't this continue to `read` while `read_bytes < length`? If data isn't available, `ret` will be set to `EAGAIN`, which should be handled in the branch checking for an error response.

No, as otherwise the caller could not know how much was read and wouldn't receive that data. Non-blocking reads have to do at most one syscall if it returns data.